### PR TITLE
Remove unnecessary logging/method info creation

### PR DESCRIPTION
**Pull request by wiktork**
Should address the race condition fixed by #385

### CODE REVIEW

File: src/InstrumentationEngine/ProfilerManager.cpp [hunk 1]
```diff
@@ -1964,18 +1953,6 @@ HRESULT CProfilerManager::JITCompilationFinished(
 
     PROF_CALLBACK_BEGIN
 
-    CComPtr<CMethodInfo> pMethodInfo;
-    hr = CreateMethodInfo(functionId, &pMethodInfo);
```
Comment (WilliamXieMSFT) [inline src/InstrumentationEngine/ProfilerManager.cpp:1968]: >CreateMethodInfo [](start = 9, length = 16)

Do we need to update the changelog with this bugfix?

**Review by jander-msft — APPROVED** [review]

**Review by WilliamXieMSFT — APPROVED** [review]

**Review by delmyers — APPROVED** [review]
:shipit:

**Pull request merged**